lib/crypto/eosdart/src/numeric.dart

```diff
@@ -236,8 +236,8 @@ Uint8List digestSuffixRipemd160(Uint8List data, String suffix) {
 
 IKey stringToKey(String s, KeyType type, int size, String suffix) {
   var whole = base58ToBinary(size + 4, s);
-  var result = IKey(type, whole);
-  var digest = digestSuffixRipemd160(result.data, suffix);
+  var result = IKey(type, whole.sublist(0,size));
```

Comment:
Why does this fix a problem and what problem does making a sublist 0, size fix? 

Why is the size a parameter?

Reply:
`size` is the number of significant bytes in the key which is apparently 32 for private & 33 for public keys (ca line 208 above). The key (as string) encodes the significant bytes plus a 4-byte checksum suffix, so the `whole` key is size+4. But when returning the key value itself from the `stringToKey` routine you have to trim off the checksum.

I have tested only with public keys in `PUB_K1...` format, should check other cases.

Reply:
Added unit test. See note in PR description; can you recommend a reference implementation for private key test data?